superset/assets/src/visualizations/core/components/createLoadableRenderer.js

```diff
@@ -0,0 +1,46 @@
+import Loadable from 'react-loadable';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  onRenderSuccess: PropTypes.func,
+  onRenderFailure: PropTypes.func,
+};
+
+const defaultProps = {
+  onRenderSuccess() {},
+  onRenderFailure() {},
+};
+
+export default function createLoadableRenderer(options) {
+  const LoadableRenderer = Loadable.Map(options);
+
+  // Extends the behavior of LoadableComponent
+  // generated by react-loadable
+  // to provide post-render listeners
+  class CustomLoadableRenderer extends LoadableRenderer {
+    componentDidMount() {
```

Comment:
related to my other comment, but what's the reasoning for adding this in `mount` + `update` instead of just `update`?

Reply:
`componentDidUpdate` will be called after 2nd, 3rd, ... `render`
but after the first `render` will only call `componentDidMount`.